index.js

```diff
@@ -59,6 +63,12 @@ async function run() {
         throw new Error('Could not login: ' + doLoginStderr);
       }
 
+      const secretPrefix = replaceSpecialCharacters(registryUri)
```

Comment:
Adding my two cents here. To avoid having to deal with this ugly output name format I propose the following changes:
1. Support a single account id (we could just pass this from the `aws-actions/configure-aws-credentials`, see my example below)
2. Allow passing the region as an input
3. Change output names to `repository-username` and `repository-password` (used dashes instead of underscores to be compatible with the convention (e.g. `skip-logout` input in this action and other actions under aws-actions use dashes as well)

```
      - name: Configure AWS credentials
        id: aws-creds-one
        uses: aws-actions/configure-aws-credentials@v1
        with:             
          aws-access-key-id: ${{ secrets.ONE_AWS_ACCESS_KEY_ID }}
          aws-secret-access-key: ${{ secrets.ONE_AWS_SECRET_ACCESS_KEY }}
          aws-region: eu-west-1

      - name: Login to Amazon ECR                 
        id: ecr-one
        uses: aws-actions/amazon-ecr-login@v2
        with:
          aws-account-id: ${{ steps.aws-creds-one.aws-account-id }}
          aws-region: eu-west-1

      - name: Configure AWS credentials
        id: aws-creds-two
        uses: aws-actions/configure-aws-credentials@v1
        with:             
          aws-access-key-id: ${{ secrets.TWO_AWS_ACCESS_KEY_ID }}
          aws-secret-access-key: ${{ secrets.TWO_AWS_SECRET_ACCESS_KEY }}
          aws-region: us-east-1

      - name: Login to Amazon ECR                 
        id: ecr-two
        uses: aws-actions/amazon-ecr-login@v2
        with:
          aws-account-id: ${{ steps.aws-creds-one.aws-account-id }}
          aws-region: eu-west-1
```

This way you can basically isolate the outputs in the context of the step and just use multiple steps to handle multiple registries in multiple accounts or regions.
Note that this is likely a breaking change and would probably deserve a version bump.

I would be more than glad to contribute these changes if this sounds like a good idea to you folks.